src/softmax.jl

```diff
@@ -33,20 +33,33 @@ julia> softmax([1, 2, 3])
 See also [`logsoftmax`](@ref).
 """
```

Comment:
May I suggest for the docstring here:
```
```jldoctest
julia> softmax([1, 2, 3])
3-element Array{Float64,1}:
  0.0900306
  0.244728
  0.665241

julia> softmax([1 2 3; 2 2 2])  # dims=1
2×3 Array{Float64,2}:
 0.268941  0.5  0.731059
 0.731059  0.5  0.268941

julia> softmax([1 2 3; 2 2 2]; dims=2)
2×3 Array{Float64,2}:
 0.0900306  0.244728  0.665241
 0.333333   0.333333  0.333333
```